docs(keyboard-key): add documentation (#4164)
twilio-labs/paste: packages/paste-website/src/pages/components/keyboard-key/index.mdx

## Examples

### Default Keyboard Key

Use Keyboard Key for a single key command. Use Keyboard Key Group for key combination commands or shortcuts. When showing a key combination, do not put a “+” between Keyboard Keys or in Keyboard Key Group.

<LivePreview scope={{ KeyboardKeyGroup, KeyboardKey, Paragraph }} language="jsx">
  {defaultExample}
</LivePreview>

### Disabled Keyboard Key

Disabled Keyboard Keys should be used when the element a shortcut activates is disabled, like a disabled Menu item or Button

<LivePreview
  scope={{ KeyboardKeyGroup, KeyboardKey, Paragraph, useMenuState, Menu, MenuButton, MenuItem, ChevronDownIcon, Box }}
  language="jsx"
  noInline
>
  {disabledExample}
</LivePreview>

### Pressed Keyboard Key

Use pressed Keyboard Keys to give visual feedback when a key is pressed.

This is especially helpful when onboarding users to keyboard shortcuts. However, use the pressed state thoughtfully, only when it enhances the user experience. It can be distracting in cases where a user is using the keyboard for other interactions, like on a page with form fields.

> The `KeyboardKeyGroup` accepts state returned from the [`useKeyCombination`](#useKeyCombination) hook that allows pressedStyling to be enabled. You must also specify `keyEvent` on the component to detect the correct key pressed.

Reviewer: ```suggestion
The `KeyboardKeyGroup` accepts state returned from the [`useKeyCombination`](#usekeycombination) hook that allows pressed styling to be enabled. You must also specify `keyEvent` on the component to detect the correct key pressed.
```

> A mapping of key events can be found [here](https://www.freecodecamp.org/news/javascript-keycode-list-keypress-event-key-codes/)

Reviewer: ```suggestion
A mapping of key events can be found [here](https://www.freecodecamp.org/news/javascript-keycode-list-keypress-event-key-codes/).
```

<LivePreview scope={{ KeyboardKeyGroup, KeyboardKey, useKeyCombination, Paragraph }} noInline language="jsx">
  {pressedExample}
</LivePreview>

> ### Keyboard Key Tooltip
> 
> The Keyboard Key Tooltip provides a way to show the keyboard shortcut associated with an element when the element is focused via the keyboard or on mouse hover.

Reviewer: ```suggestion
### Keyboard Key within Tooltip

Add Keyboard Key(s) to a [Tooltip](/components/tooltip) to show a shortcut associated with an interactive component. Use the `actionHeader` and `keyCombinationsActions` to display a heading and the associated keyboard shortcut (or shortcuts, if displaying for multiple operating systems/keyboards).
```

Reviewer: suggestion: should we add a short sentence about when we recommend using a tooltip? we prob don't want everyone using them in every possible instance

        </Td>
        <Td>
          <KeyboardKeyGroup>
            <KeyboardKey>Command</KeyboardKey>
            <KeyboardKey>B</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
      </Tr>
      <Tr>
        <Td>Italicize text</Td>
        <Td>
          <KeyboardKeyGroup>
            <KeyboardKey>Control</KeyboardKey>
            <KeyboardKey>I</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
        <Td>
          <KeyboardKeyGroup>
            <KeyboardKey>Command</KeyboardKey>
            <KeyboardKey>I</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
      </Tr>
      <Tr>
        <Td>Underline text</Td>
        <Td>
          <KeyboardKeyGroup>
            <KeyboardKey>Control</KeyboardKey>
            <KeyboardKey>U</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
        <Td>
          <KeyboardKeyGroup>
            <KeyboardKey>Command</KeyboardKey>
            <KeyboardKey>U</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
      </Tr>
    </TBody>
  </Table>
</Box>

From [W3C's guide to developing a keyboard interface](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#keyassignmentconventionsforcommonfunctions):

> The following keyboard commands should be used in any context where the actions are appropriate.

Reviewer: suggestion: can you clarify this wording a bit? it's a little confusing what you're recommending

Author: I think Sarah was trying to say that the following list of keyboard shortcuts can be used where the associated action is appropriate. So command+x can be used but only as a cut action. The paragraph below this puts it into context.

            <KeyboardKey>X</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
      </Tr>
      <Tr>
        <Td>Undo last action</Td>
        <Td>
          <KeyboardKeyGroup>
            <KeyboardKey>Control</KeyboardKey>
            <KeyboardKey>Z</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
        <Td>
          <KeyboardKeyGroup>
            <KeyboardKey>Command</KeyboardKey>
            <KeyboardKey>Z</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
      </Tr>
      <Tr>
        <Td>Redo action</Td>
        <Td>
          <KeyboardKeyGroup>
            <KeyboardKey>Control</KeyboardKey>
            <KeyboardKey>Y</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
        <Td>
          <KeyboardKeyGroup>
            <KeyboardKey>Command</KeyboardKey>
            <KeyboardKey>Shift</KeyboardKey>
            <KeyboardKey>Y</KeyboardKey>
          </KeyboardKeyGroup>
        </Td>
      </Tr>
    </TBody>
  </Table>
</Box>

### Keyboard commands to avoid

These keyboard commands should be avoided since they're used by operating systems or assistive technologies:

<List marginBottom="space60">
  <ListItem>

>     Any modifier keys + any of <KeyboardKey>Tab</KeyboardKey>, <KeyboardKey>Enter</KeyboardKey>,{" "}

Reviewer: suggestion: define modifier key

Author: Great suggestion! added

    <KeyboardKey>Space</KeyboardKey>, or <KeyboardKey>Escape</KeyboardKey>.
  </ListItem>
  <ListItem>
    <KeyboardKey>Alt</KeyboardKey> + a function key.
  </ListItem>
  <ListItem>
    <KeyboardKey>Caps Lock</KeyboardKey> + any other combination of keys.
  </ListItem>
  <ListItem>
    <KeyboardKey>Insert</KeyboardKey> + any combination of other keys.
  </ListItem>
  <ListItem>
    <KeyboardKey>Scroll Lock</KeyboardKey> + any combination of other keys.
  </ListItem>
  <ListItem>
    <KeyboardKeyGroup>
      <KeyboardKey>Control</KeyboardKey>
      <KeyboardKey>Option</KeyboardKey>
    </KeyboardKeyGroup>{" "}
    + any combination of other keys.
  </ListItem>
</List>

Read more about other conflicts with browsers and international keyboards in [W3C's guide](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#keyboardshortcuts).

### When to use Keyboard Key

> <DoDont>

Reviewer: suggestion: maybe another do/don't about using `aria-keyshortcut`?
